cpp/src/parquet/page_index.cc

```diff
@@ -636,6 +691,19 @@ class OffsetIndexBuilderImpl final : public OffsetIndexBuilder {
             page_location.__set_offset(page_location.offset + final_position);
           }
         }
+
+        /// Finalize unencoded_byte_array_data_bytes and make sure page sizes match.
+        if (offset_index_.page_locations.size() ==
+            offset_index_.unencoded_byte_array_data_bytes.size()) {
+          offset_index_.__isset.unencoded_byte_array_data_bytes = true;
```

Comment:
The check above is short hand if anything isn't provided?  we only expect two states they always match or they never match once page is added?

Reply:
Yes, it should always match if size stats is enabled. Otherwise, we should expect the list is empty.